cli/crates/server/src/servers.rs

```diff
@@ -118,12 +119,14 @@ async fn spawn_servers(
 
     let environment = Environment::get();
 
-    let bridge_handle = tokio::spawn(async move { bridge::start(bridge_port, worker_port, bridge_sender).await });
+    let mut bridge_handle =
+        tokio::spawn(async move { bridge::start(bridge_port, worker_port, bridge_sender).await }).fuse();
```

Comment:
Why is the fuse necessary? in the `tokio::select`, `poll` will only be called once, no?

Reply:
I'm not entirely sure it is needed, I added it to avoid possible panics in the `select!` macro since we are polling the same future.
Looking at the code it looks like this could happen if `Event::Reload` is received by `listener::start` before `Event::BridgeReady` is received by the first `select!`. The `bridge_handle` branch would win and the second `select!` would try to poll `bridge_handle` again. Not entirely sure if this race can happen, but if yes, even this fix is not 100% correct. Instead of a panic we could end with miniflare running, bridge server down and ignoring future `Event::Reload` events.

Reply:
ah good point, didn't think of other events when reading the review. Thanks!

Reply:
To address my last sentence, `Event::Reload` will be received by `server_loop` so it is not an issue, but for clarity I've added [8ecd9f7](https://github.com/grafbase/grafbase/pull/232/commits/8ecd9f7bec83fcb46992a43c5e0da7d2c1c0430f)